packages/amplify-cli/src/initialize-env.ts

```diff
@@ -42,31 +42,44 @@ export const initializeEnv = async (
     const categoryPluginInfoList = context.amplify.getAllCategoryPluginInfo(context);
     const availableCategories = Object.keys(categoryPluginInfoList).filter(key => initializedCategories.includes(key));
 
-    availableCategories.forEach(category => {
-      categoryPluginInfoList[category].forEach(pluginInfo => {
-        try {
-          // eslint-disable-next-line import/no-dynamic-require, global-require, @typescript-eslint/no-var-requires
-          const { initEnv } = require(pluginInfo.packageLocation);
-
-          if (initEnv) {
-            categoryInitializationTasks.push(() => initEnv(context));
-          }
-        } catch (e) {
-          context.print.warning(`Could not load initEnv for ${category}`);
+    const importCategoryPluginAndQueueInitEnvTask = async (pluginInfo, category) : Promise<void> => {
+      try {
+        const { initEnv } = await import(pluginInfo.packageLocation);
+
+        if (initEnv) {
+          categoryInitializationTasks.push(() => initEnv(context));
         }
-      });
-    });
+      } catch (e) {
+        throw new AmplifyFault('PluginNotLoadedFault', {
+          message: `Could not load plugin for category ${category}. Error: ${e.message}`,
+          resolution: `Review the error message and stack trace for additional information.`,
+          stack: e.stack,
+        });
+      }
+    };
+    for (const category of availableCategories) {
+      for (const pluginInfo of categoryPluginInfoList[category]) {
+        await importCategoryPluginAndQueueInitEnvTask(pluginInfo, category);
+      }
+    }
 
     const providerPlugins = getProviderPlugins(context);
 
     const initializationTasks: (() => Promise<$TSAny>)[] = [];
     const providerPushTasks: (() => Promise<$TSAny>)[] = [];
 
-    context.exeInfo.projectConfig.providers.forEach(provider => {
-      // eslint-disable-next-line import/no-dynamic-require, global-require, @typescript-eslint/no-var-requires
-      const providerModule = require(providerPlugins[provider]);
-      initializationTasks.push(() => providerModule.initEnv(context, amplifyMeta.providers[provider]));
-    });
+    for (const provider of context.exeInfo.projectConfig.providers) {
```

Comment:
Its good to put a check on nullable fields
```suggestion
    for (const provider of context.exeInfo?.projectConfig?.providers) {
```

Reply:
good catch!